### PR TITLE
Featured drops

### PR DESCRIPTION
**Pull request by ryanjohn1**
## Description
- Add an endpoint to return a list of featured drops

## Work Completed
- End point added, underlying worker completed along with tests

## API Changes
- listFeaturedDrops() added

## Testing
- StoreWorkerTest created with some tests and an ignored test querying devnet which confirms the data mapping works correctly

### CODE REVIEW

File: SDK/src/main/java/fan/vault/sdk/client/VaultRx.kt [hunk 1]
```diff
@@ -53,6 +53,15 @@ class VaultRx(applicationContext: Context) : VaultBase(applicationContext) {
         )
     }
 
+    /**
+     * List featured drops
+     *
+     * @return List of featured drops including associated creator data
+     */
+    fun listFeaturedDrops() = rxSingle {
```
Comment (davidbarriovault) [inline SDK/src/main/java/fan/vault/sdk/client/VaultRx.kt:61]: Will need to add coroutine equivalent function in `Vault.kt`

**Review by davidbarriovault — CHANGES_REQUESTED** [review]

**Review by anthonydmevault — CHANGES_REQUESTED** [review]

**Review by davidbarriovault — APPROVED** [review]

File: SDK/src/main/java/fan/vault/sdk/models/JsonFeaturedDrop.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package fan.vault.sdk.models
+
```
[the remark this answers is not in the document]
Comment (ryanjohn1) [reply SDK/src/main/java/fan/vault/sdk/models/JsonFeaturedDrop.kt:2]: I was a little unsure what was going on with the `/stores` endpoint. I'd heard a dedicated endpoint would exist for this at some point? So I copied what they'd done in the iOS sdk. Is this just returning all drops in the store and we have no featuring mechanism at the moment?

File: SDK/src/main/java/fan/vault/sdk/models/FeaturedDrop.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            FeaturedDrop(
+                id = variant.candyMachine,
+                dropId = dropId,
+                candyMachineId = variant.candyMachine,
+                name = variant.collectionMetadata.name,
+                image = variant.collectionMetadata.image,
+                startDate = Date(startDateSeconds * 1000),
+                price = variant.priceUsdCents,
+                creators = creators,
+                type = variant.collectionMetadata.attributes?.firstOrNull { it.traitType == "type" }?.value?.let {
```
[the remark this answers is not in the document]
Comment (ryanjohn1) [reply SDK/src/main/java/fan/vault/sdk/models/FeaturedDrop.kt:36]: I'm not sure about the store data structure. I copied the mapping from the Swift SDK

File: SDK/src/main/java/fan/vault/sdk/client/VaultRx.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryanjohn1) [reply SDK/src/main/java/fan/vault/sdk/client/VaultRx.kt:61]: Done

**Pull request merged**